src/infrastructure/redis-in-memory-sessions/cluster-add-nodes.md

```diff
@@ -38,38 +38,60 @@ After taking the steps described in this section, the architecture of your Redis
+You must perform a reshard of the cluster to redistribute the key slots evenly among all the Master processes of the Redis Cluster, including the new Master processes running on **Server 4** and **Server 5**.
 
 1. Open an SSH connection to any of the Redis Cluster machines.
 
 1. Run the following command:
 
-        ubuntu@ip-172.31.6.91:~$ redis-cli --cluster reshard 127.0.0.1:7000
+        ubuntu@[SERVER_4]:~$ redis-cli --cluster reshard 127.0.0.1:7000 -a [ACCESSKEY]
+
+1. Follow the prompts. You must configure:
+   * How many slots to move (manually defined)
+   * The receiving Master node ID
+   * The source node IDs (a specific node or all nodes)
 
 ## Step 4. Add the Replica processes to the Redis Cluster
 
-1. Add the Replica process of the first new server (Server 4, running on port 7001) to the Redis Cluster. Run the following command:
+1. Take note of the information given by the following command:
+      
+        ubuntu@[SERVER_4]:~$ redis-cli -p 7001 -a [ACCESSKEY] cluster nodes
+
+1. Add the Replica process of the first new server (**Server 4**, running on port 7001) to the Redis Cluster. Run the following command:
+
+        ubuntu@[SERVER_4]:~$ redis-cli --cluster add-node 127.0.0.1:7001 [EXISTING_CLUSTER_NODE]:7000 --cluster-slave --cluster-master-id [MASTER_NODE_ID] -a [ACCESSKEY]
+
+    * Replace `[ACCESSKEY]` with **Server 4**'s configured password.
+    * Replace `[EXISTING_CLUSTER_NODE]` with a master node's hostname/IP (for example, for **Server 1**).
+    * Replace `[MASTER_NODE_ID]` with the node ID of the Master Process on **Server 5**. 
+      * A node ID is a 40-character string (for example, `ff3e4300bec02ed4bd1be9af5d83a5b44249c2b2`) as provided in the information by the command in the first step. 
 
-        ubuntu@ip-172.31.6.91:~$ redis-cli --cluster add-node 127.0.0.1:7001 172.31.6.35:7000 --cluster-slave --cluster-master-id [MASTER_NODE_ID]
+1. Repeat the last command to setup the Replica process running on **Server 5**:
 
-    Replace `[MASTER_NODE_ID]` with the node ID of the Master Process you want to replicate (in this case, the Master process on Server 5). A node ID is a 40-character string (for example, `ff3e4300bec02ed4bd1be9af5d83a5b44249c2b2` for the Master process running on Server 1).
+        ubuntu@[SERVER_5]:~$ redis-cli --cluster add-node 127.0.0.1:7001 [EXISTING_CLUSTER_NODE]:7000 --cluster-slave --cluster-master-id [MASTER_NODE_ID] -a [ACCESSKEY]
 
-1. Repeat the last command to setup the Replica process running on Server 5. In this case, set `[MASTER_NODE_ID]` to the node ID of the Master process running on Server 4.
+    * Replace `[ACCESSKEY]` with **Server 5**'s configured password. 
+    * Replace `[EXISTING_CLUSTER_NODE]` with a master node's hostname/IP (for example, for **Server 1**).
+    * Replace `[MASTER_NODE_ID]` with the node ID of the Master Process on **Server 4**.
+      * Again, refer to the information provided in the command in the first step.
```

Comment:
Why? What do we refer to the information for?

Reply:
This [Master_Node ID] value needs to be replaced by the value copied from the "1. Take note of the information given by the following command:"

Reply:
This is to avoid the same but basically it is to do the same as the previous step but replacing the `[MASTER_NODE_ID]` with the node ID of the Master Process on **Server 4**. This is id should be retrieved using the command in the first step.